example_test.go

```diff
@@ -153,63 +153,26 @@ func ExampleSign1() {
 	}
 
 	// sign message
-	protected := cose.ProtectedHeader{}
-	protected.SetAlgorithm(cose.AlgorithmES512)
-	msg, err := cose.Sign1(rand.Reader, signer, protected, []byte("hello world"), nil)
-	if err != nil {
-		panic(err)
+	headers := cose.Headers{
+		Protected: cose.ProtectedHeader{
+			cose.HeaderLabelAlgorithm: cose.AlgorithmES512,
+		},
+		Unprotected: cose.UnprotectedHeader{
+			cose.HeaderLabelKeyID: 1,
+		},
 	}
-
-	// update unprotected headers
-	msg.Headers.Unprotected[cose.HeaderLabelKeyID] = 1
-
-	// encode message
-	sig, err := msg.MarshalCBOR()
+	sig, err := cose.Sign1(rand.Reader, signer, headers, []byte("hello world"), nil)
 	if err != nil {
 		panic(err)
 	}
+
 	fmt.Println("message signed")
-	_ = sig // futher process on sig
+	_ = sig // further process on sig
 	// Output:
 	// message signed
 }
 
-// This example demonstrates verifying COSE_Sign1 signatures using Verify1().
-func ExampleVerify1() {
-	// get a signed message and a trusted public key
-	sig, publicKey := getSignatureAndPublicKey()
-
-	// create a verifier from a trusted public key
-	verifier, err := cose.NewVerifier(cose.AlgorithmES512, publicKey)
-	if err != nil {
-		panic(err)
-	}
-
-	// verify message
-	var msg cose.Sign1Message
-	err = msg.UnmarshalCBOR(sig)
-	if err != nil {
-		panic(err)
-	}
-	err = cose.Verify1(&msg, nil, verifier)
-	if err != nil {
-		panic(err)
-	}
-	fmt.Println("message verified")
-
-	// tamper the message and verification should fail
-	msg.Payload = []byte("foobar")
-	err = cose.Verify1(&msg, nil, verifier)
-	if err != cose.ErrVerification {
-		panic(err)
-	}
-	fmt.Println("verification error as expected")
-	// Output:
-	// message verified
-	// verification error as expected
-}
-
-// getSignatureAndPublicKey is a helping function for ExampleVerify1().
+// getSignatureAndPublicKey is a helping function for ExampleSign1Message_Verify().
 func getSignatureAndPublicKey() ([]byte, crypto.PublicKey) {
```

Comment:
We can safely remove this `getSignatureAndPublicKey()` function since it is not used anywhere.

Reply:
Removed!